framework_tvb/tvb/core/services/import_service.py

```diff
@@ -409,6 +409,15 @@ def load_datatype_from_file(self, current_file, op_id, datatype_group=None, curr
         """
         self.logger.debug("Loading DataType from file: %s" % current_file)
         h5_class = H5File.h5_class_from_file(current_file)
+        reference_list = h5_class(current_file).gather_references()
```

Comment:
Is this happening late enough in the flow to work correctly in case of a full project import (where multiple linked dts are being imported together)?

Reply:
Hmm, not really. I tested it and if there is a missing referenced datatype, then it doesn't import the project. I guess the project should still be imported, only the datatypes with missing references should be skipped, right?

Reply:
Yes, for partially correct projects we should do a best effort to import what we can, and show a warning that not everything was imported.

What I had in mind was a project PR1 with Conn1, Surface2 and RegMapp (referencing Conn1 and Surface2) datatypes to be imported. Will this import work (if we have an empty db before)?
Could be tested also with the current DefaultProject this situation.

Reply:
After seeing the entire File again, i would answer this myself: 
in case of a project import with linked dts, we order by operation create date, iterate these ops, store in db all linked dts for one op, then we continue with the next operation. The check recently added by Robert about linked dts happens only late, when we are about to load a dt within the ops loop(previous ops were already stored).
Still we might have a problem if inside one operation folder we have linked dts (both conn1 and regionMapping are in the same operation generated). Could you validate/ check this situation, please?

Reply:
It doesn't work for that cases, because at that point the datatypes are not taken in the order of their creation. I tried an example with a RegionVolumeMapping depending on a Volume and the RVM came before the Volume so the import failed even though the Volume was there.

Reply:
But where could I add the check if not when it iterates through the operations?

Reply:
Sorting based on the creation date should be enough also here, but I think it should be done in `_load_datatypes_from_operation_folder`, could that code be extracted and reused in both places (migration and import)?